bot/slack/run_async.py

```diff
@@ -0,0 +1,33 @@
+
+def run_async(func):
+    """
+    run_async(func)
+    function decorator, intended to make "func" run in a separate
+    thread (asynchronously).
+    Returns the created Thread object
+    from: http://code.activestate.com/recipes/576684-simple-threading-decorator/
+
+    E.g.:
+    @run_async
+    def task1():
+    do_something
+
+    @run_async
+    def task2():
+    do_something_too
+
+    t1 = task1()
+    t2 = task2()
+    ...
+    t1.join()
+    t2.join()
+    """
+    from threading import Thread
+    from functools import wraps
+
+    @wraps(func)
+    def async_func(*args, **kwargs):
```

Comment:
I'm not a fan of the idea of using a decorator here.  This function does not modify its function argument at all, but rather has side-effects, which makes for a leaky and obtuse abstraction.

While requiring more code, we should simply write the three lines of this function's body where calls to an async function occur rather than hiding its behavior this way.

Reply:
This was a copy/pasta from the original IRC mozdefbot, so could you explain this a little more of what you're asking please? I'm honestly not entirely sure what `@wraps` is doing here.

Reply:
I think it was trying to add some threading to work around timing issues in the kitnirc client. If we aren't using that anymore you can probably ditch it

Reply:
I'm basically just saying that instead of writing

```py
@async_run
def function(...):
  # ...
```

we should prefer to write

```py
handler = Thread(target=function, ...)
handler.start()
...
```

directly.

Reply:
I ended up removing the run_async file and implemented it like you recommended.